packages/x-data-grid/src/components/columnHeaders/ColumnHeaderMenuIcon.tsx

```diff
@@ -46,11 +47,19 @@ export const ColumnHeaderMenuIcon = React.memo((props: ColumnHeaderMenuIconProps
   );
 
   const columnName = colDef.headerName ?? colDef.field;
+  const titleWidth =
+    divRef.current?.parentElement?.firstElementChild?.firstElementChild?.clientWidth;
+  const neededWidth =
+    divRef.current?.parentElement?.firstElementChild?.firstElementChild?.firstElementChild
+      ?.scrollWidth ?? 1;
```

Comment:
this can break with custom header renderers, where the first element isn't necessarily a text block, or there are follow-up elements like in the docs example: [Custom header renderer](https://next.mui.com/x/react-data-grid/column-header/#custom-header-renderer)